### DISCUSSION TITLE
ServiceStack.Redis ACL ISSUE

### DISCUSSION THREAD

**Discussion by wurangbo** [ServiceStack/Discuss#124] [Q&A] [answered]
Hi guys,
    When I Access redis with ACL Mode， The following exception was thrown：
    InnerException = {"WRONGPASS invalid username-password pair or user is disabled."}
    ServiceStack.Redis.RedisResolver.CreateRedisClient(RedisEndpoint config, Boolean master)
    ServiceStack.Redis.RedisResolver.CreateMasterClient(Int32 desiredIndex)
    ServiceStack.Redis.RedisResolverExtensions.CreateMasterClient(IRedisResolver resolver, Int32 desiredIndex)
    ServiceStack.Redis.PooledRedisClientManager.GetClientBlocking()
    ServiceStack.Redis.PooledRedisClientManager.GetClient()

     connStr : <add key="DefaultReadRedisServer" value="siteweb:siteweb1123!@10.163.96.61:6379"/>

**Comment by mythz** [comment]
Try adding the password URL Encoded on the queryString, see the [Redis Connection Strings docs](https://docs.servicestack.net/redis/client-managers#redis-connection-strings), e.g:

```csharp
var connString = $"redis://{Host}?ssl=true&username={Username}&password={Password.UrlEncode()}";
```

**Comment by wurangbo** [comment]
Hi Guys,
Use the following connstr access according to the instructions, but it still doesn't work
<add key="DefaultReadRedisServer" value="redis://10.163.96.61:6379"?ssl=true&username=siteweb&password=siteweb123!"/>

And now I need a complete connstr example 
 connStr :<add key="DefaultReadRedisServer" value="redis://10.163.96.61:6379"?ssl=true&username=siteweb&password=siteweb123!"/>
-->  ??
thanks

**Reply by mythz** [reply, marked as answer]
Your xml has a double quote making it malformed and you need to xml encode values, try:

```xml
<add key="DefaultReadRedisServer" value="redis://10.163.96.61:6379?ssl=true&amp;username=siteweb&amp;password=siteweb123!"/>
```

**Comment by wurangbo** [comment]
Great!
